app/models/whitehall_migration/asset_import.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Represents the raw import of an asset from Whitehall Publisher and
+# the import status of the asset into Content Publisher
+class WhitehallMigration::AssetImport < ApplicationRecord
+  belongs_to :document_import
+  belongs_to :image_revision, class_name: "Image::Revision", optional: true
+  belongs_to :file_attachment_revision, class_name: "FileAttachment::Revision", optional: true
+
+  validate :associated_with_only_image_or_file_attachment
+
+  def main_asset
+    if image_revision.present?
+      image_revision.asset("960")
+    else
+      file_attachment_revision.asset
+    end
+  end
```

Comment:
I wonder if this method should be content_publisher_asset and be able to determine what based on the variant.

For example if we have a variant of original (or nil) we return `image_revision.asset("960")` or `file_attachment_revision.asset` then after that we have things like s960 and similar where they may return an image_revision asset. Then things like thumbnail or unsupported Content Publisher sizes where we'd return nil.

Reply:
So you're proposing something like this?

```ruby
def content_publisher_asset
  if variant.nil?
    image_revision.asset("960") || file_attachment_revision.asset
  elsif file_attachment_revision.present?
    # No content publisher equivalents for file attachment variants
    nil
  else
    case image_revision.variant
    when "s960", "s712", "s630"
      image_revision.asset("960")
    when "s465", "s300", "s216"
      image_revision.asset("300")
    else
      # unsupported file size
      nil
    end
  end
end
```

...and in the migrate assets script, we could then simplify to something like:

```diff
if whitehall_asset.main_asset.live? &&
-    (whitehall_asset.variant.nil? || whitehall_asset.image_revision.present?)
+    (whitehall_asset.content_publisher_asset.present?)
  GdsApi.asset_manager.update_asset(
    whitehall_asset.whitehall_asset_id,
-    redirect_url: whitehall_asset.main_asset.file_url,
+    redirect_url: whitehall_asset.content_publisher_asset,
  )
else
  GdsApi.asset_manager.delete_asset(whitehall_asset.whitehall_asset_id)
end
```

Reply:
Yup 👍